test/xbgpu/test_engine.py

```diff
@@ -501,6 +501,17 @@ async def test_xengine_end_to_end(
         boundary to more accurately test the setting of the first output
         packet's timestamp (to be non-zero).
         """
+        # NOTE: `HEAPS_PER_FENGINE_PER_CHUNK` and the `heap_accumulation_threshold`s
+        # are chosen carefully for this test. We simulate the first accumulation
+        # being incomplete by ensuring there is only one chunk of data present.
+        # In doing so, we must ensure `HEAPS_PER_FENGINE_PER_CHUNK` is smaller
+        # than either pipeline's `heap_accumulation_threshold`. The logic of the
+        # test has *not* been verified if that constraint isn't met, it merely
+        # `assert`'s.
+        heap_acc_thresholds = [corrprod_output.heap_accumulation_threshold for corrprod_output in corrprod_outputs]
+        for heap_accumulation_threshold in heap_acc_thresholds:
+            assert heap_accumulation_threshold > HEAPS_PER_FENGINE_PER_CHUNK
```

Comment:
Could also do
```suggestion
        assert min(heap_accumulation_thresholds) > HEAPS_PER_FENGINE_PER_CHUNK
```
but that's just bikeshedding - feel free to keep it as is if it seems easier to understand that way.

Reply:
Argh, too right. I think your suggestion is easier and/or exactly what I had in mind (just couldn't think of the syntax).